packages/hardhat-core/test/fixture-projects/compilation-import-folder/contracts/A.sol

```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.19;
+
+// The file has to not exist for test purposes
+import "some-plugin"
```

Comment:
This is very minor, but "plugin" normally refers to Hardhat plugins, not to solidity libraries. So I would rename this to `some-lib` and would remove the `index.js` from `node_modules/some-plugin`.

Also, I think we should have a different test that imports a directory too but using a relative path.